pkg/sdk/go/keys.go

```diff
@@ -6,43 +6,45 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"time"
 
 	"github.com/mainflux/mainflux/pkg/errors"
 )
 
 const keysEndpoint = "keys"
 
-func (sdk mfSDK) Issue(token string, k Key) (issueKeyRes, error) {
-	data, err := json.Marshal(k)
+func (sdk mfSDK) Issue(token string, d time.Duration) (KeyRes, error) {
+	datareq := KeyReq{Type: 2, Duration: d}
```

Comment:
Magic number `2` - define all constants at the top of the file and add appropriate comments.